File: src/main/com/intellij/lang/jsgraphql/schema/builder/GraphQLDirectiveTypeCompositeDefinition.java
package com.intellij.lang.jsgraphql.schema.builder;

import com.intellij.lang.jsgraphql.types.language.DirectiveDefinition;
import com.intellij.util.containers.ContainerUtil;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class GraphQLDirectiveTypeCompositeDefinition extends GraphQLCompositeDefinition<DirectiveDefinition> {

  @Override
  protected @NotNull DirectiveDefinition mergeDefinitions(@NotNull List<DirectiveDefinition> sourceDefinitions) {
    DirectiveDefinition definition = ContainerUtil.getFirstItem(sourceDefinitions);
    return definition.transform(builder -> builder.sourceNodes(sourceDefinitions));
  }
}

File: src/main/com/intellij/lang/jsgraphql/schema/builder/GraphQLDirectiveTypeCompositeDefinition.kt
package com.intellij.lang.jsgraphql.schema.builder

import com.intellij.lang.jsgraphql.types.language.DirectiveDefinition

class GraphQLDirectiveTypeCompositeDefinition : GraphQLCompositeDefinition<DirectiveDefinition>() {
  override fun mergeDefinitions(sourceDefinitions: List<DirectiveDefinition>): DirectiveDefinition {
    return sourceDefinitions.first().transform { builder -> builder.sourceNodes(sourceDefinitions) }
  }
}
